### PR TITLE
Exchange Rate Caching

### PR DESCRIPTION
**Pull request by shahthepro**
It seems like the API call that is made to fetch exchange rate is rate-limited and fails some times. Instead of invoking that API every time, we can cache it in Redis/DB and poll it in fixed intervals. We can have all requests query a single endpoint on bridge that returns the cached exchange rate for that crypto-fiat pair.

### CODE REVIEW

**Comment by shahthepro** [comment]
@tomlinton As of now, Polling only works for ETH/USD market. I just wanted to know what you think about this before writing a more generic version of the poller

**Comment by tomlinton** [comment]
Thanks for doing that so quickly @shahthepro! It looks great. Only thing I add is maybe adjust a few of the logger calls to warnings or errors where appropriate as they are all `logger.log`. It might make it easier to read the logs down the track.

I'll just add a bit of context here for anyone wondering what this is about. We were seeing `cryptonator.com` not responding to requests for some reason (perhaps rate limiting, perhaps just issues at their end, more [discussion here](https://github.com/OriginProtocol/origin/issues/228)). If that happens and the Origin marketplace app can not retrieve any exchange rate it'll crash when trying to convert crypto to fiat values. @shahthepro and I discussed and decided it'd be good to add our own exchange rate endpoints to the bridge as its a lot easier to ensure our own server works, and we won't have to push new builds to the App/Play stores if we need to swap out the exchange rate API.

**Comment by franckc** [comment]
I like the idea. Only one question: have you considered putting this in the graphql server rather than bridge ? Bridge currently specializes in handling attestation related methods. Currency conversion feels more like business logic... The drawback of putting this in graphql is that it would need to run only on server side and not in dapp when performance mode is disabled...
I'm just asking to make sure you considered that option - I don't feel super strongly. If it is much simpler to implement in bridge, totally fine to go that route.

**Comment by tomlinton** [comment]
I don't really have any strong feelings either. I think of bridge as a bridge between any third party APIs which this fits nicely into. I'd prefer we just have a single source of truth for it though - as you noted it gets tricky if this is running in GraphQL client side and we start seeing the failures again.

File: infra/bridge/src/controllers/utils.js [hunk 1]
```diff
@@ -34,4 +36,13 @@ router.post('/exists', async (req, res) => {
   res.end()
 })
 
+/**
+ * Returns excahnge rate of given market
```
Comment (tomlinton) [inline infra/bridge/src/controllers/utils.js]: Typo

**Comment by shahthepro** [comment]
@franckc 

> have you considered putting this in the graphql server rather than bridge? 

No, I also thought that bridge was to be used with external API services. If we were to add this in GraphQL server instead, we can probably use Apollo's InMemory cache instead of Redis. But still we may run into issues when not in performance mode as you say.

**Pull request merged**